src/fp.rs

```diff
@@ -422,7 +426,7 @@ pub(crate) mod tests {
         }
 
         fn radix(&self) -> BigInt {
-            BigInt::from(1) << 128
+            BigInt::from(self.base).pow(2)
```

Comment:
Is the base always a power of 2? If so, is this worth it? I suppose it's possible that `BigInt` 
```suggestion
            debug_assert_eq!(self.base, (self.base - 1).next_power_of_two())
            BigInt::from(self.base) << 1
```

Reply:
Alternative to `debug_assert_eq!()`: move this to unit tests.

Reply:
good point, added one test